bcda/lambda/cclf/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/sirupsen/logrus"
+
+	bcdaaws "github.com/CMSgov/bcda-app/bcda/aws"
+	"github.com/CMSgov/bcda-app/bcda/cclf"
+	"github.com/CMSgov/bcda-app/optout"
+
+	"github.com/CMSgov/bcda-app/conf"
+)
+
+func main() {
+	if os.Getenv("LOCAL_STACK_ENDPOINT") != "" {
+		res, err := handleCclfImport(os.Getenv("BFD_BUCKET_ROLE_ARN"), os.Getenv("BFD_S3_IMPORT_PATH"))
+		if err != nil {
+			fmt.Printf("Failed to run opt out import: %s\n", err.Error())
+		} else {
+			fmt.Println(res)
+		}
+	} else {
+		lambda.Start(cclfImportHandler)
+	}
+}
+
+func cclfImportHandler(ctx context.Context, sqsEvent events.SQSEvent) (string, error) {
+	env := conf.GetEnv("ENV")
+	appName := conf.GetEnv("APP_NAME")
+	logger := configureLogger(env, appName)
+
+	s3Event, err := bcdaaws.ParseSQSEvent(sqsEvent)
+
+	if err != nil {
+		logger.Errorf("Failed to parse S3 event: %v", err)
+		return "", err
+	} else if s3Event == nil {
+		logger.Info("No S3 event found, skipping safely.")
+		return "", nil
+	}
+
+	for _, e := range s3Event.Records {
+		if e.EventName == "ObjectCreated:Put" {
+			s3AssumeRoleArn, err := loadBfdS3Params()
+			if err != nil {
+				return "", err
+			}
+
+			parts := strings.Split(e.S3.Object.Key, "/")
+
+			if len(parts) == 1 {
+				return handleCclfImport(s3AssumeRoleArn, e.S3.Bucket.Name)
+			} else {
+				directory := fmt.Sprintf("%s/%s", e.S3.Bucket.Name, parts[0])
+				return handleCclfImport(s3AssumeRoleArn, directory)
+			}
+		}
+	}
+
+	logger.Info("No ObjectCreated:Put events found, skipping safely.")
+	return "", nil
+}
+
+func loadBfdS3Params() (string, error) {
+	env := conf.GetEnv("ENV")
+
+	bcdaSession, err := bcdaaws.NewSession("", os.Getenv("LOCAL_STACK_ENDPOINT"))
```

Comment:
comment from the other PR applies here, too, I think this NewSession function may be the spot to consolidate the explanation?